### DISCUSSION TITLE
[coc.nvim] Your system isn't built by our CI!

### DISCUSSION THREAD

**Discussion by tigerinus** [zigtools/zls#664] [Q&A] [answered]
Tried to enable coc-zls in neovim, but keep getting following error after `nvim main.zig`:

![image](https://user-images.githubusercontent.com/7172560/190935464-7d64537a-36ae-4085-b7dc-2f568b63f880.png)

Type `y`

![image](https://user-images.githubusercontent.com/7172560/190935477-0bcc1dcb-206f-42c0-a1b7-f4b21e6709e7.png)

My `init.vim` has:

```vim
" vim-plug
call plug#begin('~/.vim/plugins')
Plug 'neoclide/coc.nvim', {'branch': 'release'} " language server support
Plug 'ziglang/zig.vim'                          " zig
...
```

My `zls.json`:

```json
cat  /home/ubuntu/.config/zls.json
{
    "$schema": "https://raw.githubusercontent.com/zigtools/zls/master/schema.json",
    "zig_exe_path": "/home/ubuntu/.local/bin/zig",
    "enable_snippets": true,
    "enable_ast_check_diagnostics": true,
    "enable_import_embedfile_argument_completions": true,
    "warn_style": true,
    "enable_semantic_tokens": true,
    "enable_inlay_hints": true,
    "operator_completions": true,
    "include_at_in_builtins": false,
    "max_detail_length": 1048576
}
```

**Comment by phan-do** [comment]
Got this error too. Seems to be a problem with `coc-zls`, not ZLS itself. Specifically, [this block of code](https://github.com/xiyaowong/coc-zls/blob/6cc6d8926042df86631127eed27fe1985987046e/src/zls.ts#L29-L41.), which deals with finding the default installation path for (presumably) ZLS. It doesn't have any logic to handle aarch64-darwin systems sooooo I'm guessing that's why.

**Comment by SuperAuguste** [comment, marked as answer]
Yep, this is correct; `coc-zls` is a lightish fork of `zls-vscode` and that specific portion isn't up to date [(see the updated version)](https://github.com/zigtools/zls-vscode/blob/master/src/extension.ts#L33-L50). Please file an issue or (better yet) a PR in the `coc-zls` repo; we can't do much about it from here :)
